### PR TITLE
MISRA compliance changes

### PR DESCRIPTION
**Pull request by dan4thewin**
Addresses many of the MISRA findings from Coverity

### CODE REVIEW

File: libraries/standard/json/include/json.h [hunk 1]
```diff
@@ -70,7 +70,7 @@ JSONStatus_t JSON_Validate( const char * buf,
  * may descend through nested objects when the queryKey contains matching
  * key strings joined by a separator.
  *
- * For example, if buf contains '{"foo":"abc","bar":{"foo","xyz"}}', then a 
+ * For example, if buf contains '{"foo":"abc","bar":{"foo","xyz"}}', then a
```
Comment (yourslab) [inline libraries/standard/json/include/json.h:73]: Is {"foo":"abc","bar":{"foo","xyz"}} valid JSON?
Did you mean {"foo":"abc","bar":{"foo":"xyz"}}?

**Comment by sarenameas** [comment]
"coverity" needs to be added to lexicon.txt

File: libraries/standard/json/include/json.h (hunk 1 above)
Comment (sarenameas) [reply libraries/standard/json/include/json.h:73]: Or maybe {"foo":"abc","bar":["foo","xyz"]}

Comment (dan4thewin) [reply libraries/standard/json/include/json.h]: Oops.  Fixed.  Thank you!

**Pull request merged**